sthree/bucket.go

```diff
@@ -676,27 +679,46 @@ func (b *Bucket) SyncTo(local, prefix string, withDelete bool) error {
 	return nil
 }
 
+// SyncOptions controls the behavior of the sync-to and sync-from operations.
+type SyncOptions struct {
+	WithDelete         bool
+	Timeout            time.Duration
+	QueueCheckInterval time.Duration
+}
+
+// NewDefaultSyncOptions populates a SyncOptions object with reasonable defaults.
+func NewDefaultSyncOptions() SyncOptions {
+	return SyncOptions{
+		Timeout:            10 * time.Minute,
```

Comment:
IIUC, I'd argue that 10 minutes is too much. A regular push task may copy about 10 packages, and if all sync operations time out at 10 minutes we still have to wait 100 minutes for results -- which is not far from the 120min we wait for a task to time out now.

This change LGTM as it paves the way to make this configurable via a parameter later, so feel free to check it in with these values and we'll iterate on what the default should be.

Reply:
It's a 10 minute timeout for all packages in the sync from (way more than 10) and then a separate 10m timeout for the final sync operation. If you hit either the operation aborts. So it turns a 2 hour timeout into a 10m timeout. 

What I'm less aware of is "does this process always complete in less than <10 (or maybe 12) minutes when it's going to be successful".